File: Phan_B/baitap45/baitap45.jl
function luy_thua_mod(a::Int64, b::Int64, m::Int64)
    r = Int64(1 % m)
    a = a % m
    while b > 0
        if (b & 1) == 1
            r = (r * a) % m
        end
        a = (a * a) % m
        b >>= 1
    end
    return r
end

function main()
    content = read(stdin, String)
    tokens = split(content)
    if isempty(tokens) return end

    idx = 1
    while idx <= length(tokens)
        a = parse(Int64, tokens[idx])
        b = parse(Int64, tokens[idx + 1])
        m = parse(Int64, tokens[idx + 2])
        idx += 3
        println(luy_thua_mod(a, b, m))
        flush(stdout)
    end
end

main()

File: Phan_B/baitap45/baitap45.py
import sys

def luy_thua_mod(a: int, b: int, m: int) -> int:
    r = 1 % m
    a %= m
    while b > 0:
        if b & 1:
            r = (r * a) % m
        a = (a * a) % m
        b >>= 1
    return r

def main():
    content = sys.stdin.read().split()
    if not content:
        return
    idx = 0
    while idx < len(content):
        a = int(content[idx])
        b = int(content[idx + 1])
        m = int(content[idx + 2])
        idx += 3
        print(luy_thua_mod(a, b, m), flush=True)

if __name__ == "__main__":
    main()